components/coda/actions/list-columns/list-columns.mjs

```diff
@@ -30,33 +29,37 @@ export default {
         "visibleOnly",
       ],
     },
-    limit: {
+    max: {
       propDefinition: [
         coda,
-        "limit",
-      ],
-    },
-    pageToken: {
-      propDefinition: [
-        coda,
-        "pageToken",
+        "max",
       ],
     },
   },
   async run({ $ }) {
     let params = {
       visibleOnly: this.visibleOnly,
-      limit: this.limit,
-      pageToken: this.pageToken,
     };
 
-    let response = await this.coda.listColumns(
-      this.docId,
-      this.tableId,
-      params,
-    );
+    let items = [];
+    let response;
+    do {
+      response = await this.coda.listColumns(
+        $,
+        this.docId,
+        this.tableId,
+        params,
+      );
+      items.push(...response.items);
+      params.pageToken = response.nextPageToken;
+    } while (params.pageToken && items.length < this.max);
 
-    $.export("$summary", `Retrieved ${response.items.length} column(s)`);
-    return response;
+    if (items.length > this.max) items.length = this.max;
+
+    $.export("$summary", `Retrieved ${items.length} column(s)`);
+
+    return {
+      items,
+    };
```

Comment:
The return value from a "list" action is typically an array of the items being listed, to simplify using the results from actions in workflows. Some examples in existing components: [Airtable](https://github.com/PipedreamHQ/pipedream/blob/78b07a8bc24adb1724467987e85743fa7bed711c/components/airtable/actions/common-list.js#L54), [Google Sheets](https://github.com/PipedreamHQ/pipedream/blob/1dcc47febdc68b2b9d1990c3d7658f91a0d1f090/components/google_sheets/actions/list-worksheets/list-worksheets.mjs#L30), [Eventbrite](https://github.com/PipedreamHQ/pipedream/blob/78b07a8bc24adb1724467987e85743fa7bed711c/components/eventbrite/actions/get-event-attendees/get-event-attendees.js#L41)

```suggestion
    return items;
```

Also applies to `list-docs.mjs` and `list-tables.mjs`.

Reply:
Addressed